python/mxnet/ndarray/ndarray.py

```diff
@@ -387,19 +388,15 @@ def __setstate__(self, state):
         else:
             self.handle = None
 
-    # pylint: disable=line-too-long
     def __setitem__(self, key, value):
         """x.__setitem__(i, y) <=> x[i]=y
 
-        Sets value to self[key]. This functions supports advanced indexing defined in the following reference with
-        some restrictions.
-
-        https://docs.scipy.org/doc/numpy-1.13.0/reference/arrays.indexing.html#combining-advanced-and-basic-indexing
+        Sets ``self[key]`` to ``value``.
 
-        - If key is a list type, only a list of integers is supported, e.g. key=[1, 2] is supported,
```

Comment:
We only supported list of integers when the `key` is a list because we find the behavior of nested lists are confusing. For example, `x[[1], [2]]` is equivalent to `x[[[1], [2]]]`. Has this implementation enable indexing with key as a nested list? If so, could you explain and document the behavior consistent with numpy?

Reply:
The whole topic of advanced indexing is deeply confusing, and I've heard people say that it's too advanced to be understandable (and useful).
Your second example, though, is deprecated and will change in the future (to be aware of here as well!). If wrapping the whole index in a sequence, a `tuple` must be used, not a `list`. Here's the warning from current Numpy:

    FutureWarning: Using a non-tuple sequence for multidimensional indexing is
    deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be
    interpreted as an array index, `arr[np.array(seq)]`, which will result either in an
    error or a different result.

Re: compatibility with Numpy: within the reach of the current unit tests, I think advanced indexing is 100 % compatible with Numpy's. So I guess we can just point to the Numpy documentation for a detailed explanation.
The only remaining difference is with respect to strided views, where Numpy has more cases (in basic indexing) where a view is returned. Getting on par there would require a much bigger effort (extend storage backend).